lib/ComposerInspector/ComposerInspector.php

```diff
@@ -7,57 +7,73 @@
 
 final class ComposerInspector
 {
+    private const DEFAULT_BIN_DIR = 'vendor/bin';
+
     /**
      * @var array<string,Package>
      */
     private array $packages = [];
 
+    private string $binDir = self::DEFAULT_BIN_DIR;
+
     private bool $loaded = false;
 
-    public function __construct(private string $path)
-    {
+    public function __construct(
+        private string $lockFile,
+        private string $composerFile,
+    ) {
     }
 
     public function package(string $name): ?Package
     {
-        $this->readFile();
+        $this->readFiles();
         if (!isset($this->packages[$name])) {
             return null;
         }
 
         return $this->packages[$name];
     }
 
-    private function readFile(): void
+    public function getBinDir(): string
+    {
+        $this->readFiles();
+        return $this->binDir;
+    }
+
+    private function readFiles(): void
     {
-        if ($this->loaded) {
+        if ($this->loaded || !file_exists($this->lockFile) || !file_exists($this->composerFile)) {
+            $this->loaded = true;
```

Comment:
both files are now optional...

Reply:
Yes, before the composer file was optional and the lock file should also be optional, because you don't have to install the dependencies to use this feature.

Reply:
I mean you could have a JSON file but not a lock file or vice versa potentially but this requires thatbyiu have both. We should allow it to work with only a JSON file